pkg/operator/ceph/disruption/machinelabel/add.go

```diff
@@ -0,0 +1,65 @@
+package machinelabel
+
+import (
+	mapiv1 "github.com/openshift/cluster-api/pkg/apis/machine/v1beta1"
+	cephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	"github.com/rook/rook/pkg/operator/ceph/disruption/controllerconfig"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+const (
+	osdPodLabelKey         = "app"
+	osdPODLabelValue       = "rook-ceph-osd"
+	osdClusterNameLabelKey = "rook_cluster"
+)
+
+func Add(mgr manager.Manager, opts *controllerconfig.Options) error {
+	mgrScheme := mgr.GetScheme()
+	cephv1.AddToScheme(mgrScheme)
+	mapiv1.AddToScheme(mgrScheme)
+
+	reconcileMachineLabel := &ReconcileMachineLabel{
+		client:  mgr.GetClient(),
+		scheme:  mgrScheme,
+		options: opts,
+		log:     log,
+	}
+
+	reconciler := reconcile.Reconciler(reconcileMachineLabel)
+	// create a new controller
+	c, err := controller.New(controllerName, mgr, controller.Options{Reconciler: reconciler})
+	if err != nil {
+		return err
+	}
+
+	// Watch for MachineDisruptionBudgets and enqueue the CephCluster in the namespace
+	return c.Watch(&source.Kind{Type: &corev1.Pod{}}, &handler.EnqueueRequestsFromMapFunc{
```

Comment:
This Reconciler is only triggered by Pod events. It should also be triggered by events on machines with owner references.

Reply:
yeah added a watch for that